includes/class-loc-config.php

```diff
@@ -200,7 +236,37 @@ public static function map_provider_callback(array $args) {
 		echo '<select name="' . $name . '">';
 		echo '<option value="OSM" '  . selected( $text, 'OSM' ) .  '>' . __( 'OpenStreetMap/MapBox', 'simple-location' ) . '</option>';
 		echo '<option value="Google" '  . selected( $text, 'Google' ) .  '>' . __( 'Google Maps', 'simple-location' ) . '</option>';
-		echo '<option value="Bing" '  . selected( $text, 'Bing' ) .  '>' . __( 'Bing Maps', 'simple-location' ) . '</option>';	
+		echo '<option value="Bing" '  . selected( $text, 'Bing' ) .  '>' . __( 'Bing Maps', 'simple-location' ) . '</option>';
+		echo '</select><br /><br />';
+	}
+
+	public static function mapbbox_style_callback ( array $args ) {
+		$name = $args['name'];
+		$text = get_option( $name );
+		$mapboxuser = get_option( 'sloc_mapbox_user' );
+		$api = get_option( 'sloc_mapbox_api' );
+		$url = 'https://api.mapbox.com/styles/v1/' . $mapboxuser . '?access_token=' . $api;
+		$request = wp_remote_get( $url );
+
+		if ( is_wp_error( $request ) ) {
+			return false; // Bail early.
+		}
+		$body = wp_remote_retrieve_body( $request );
+
+		$data = json_decode( $body );
+
+		echo '<select name="' . $name . '">';
```

Comment:
The styles should be separated from the style generating code. Probably by a function that takes an array of values and turns it into an options list. That will allow it to be used for styles in any map provider. The code to retrieve an array of styles should then move into the map provider class